pyteal/compiler/optimizer/optimizer.py

```diff
@@ -0,0 +1,73 @@
+from typing import List, cast
+
+from pyteal.ir.tealcomponent import TealComponent
+from pyteal.ir.tealop import TealOp
+from pyteal.ir.ops import Op
+
+
+class OptimizeOptions:
+    """An object which specifies optimizations to be performed."""
+
+    use_slot_to_stack: bool
+    iterate_optimizations: bool
+
+    def __init__(
```

Comment:
@algoidurovic Optional and minor:  If you'd prefer to save the boilerplate of initializing `OptimizeOptions`, I think we can begin using https://docs.python.org/3/library/dataclasses.html.

We've already discussed picking up the 3.10 version bump (https://github.com/algorand/pyteal/issues/244).  So, if you prefer it, I don't see harm in using a 3.7 feature.